client/src/features/GamePage/components/Layout/Layout.tsx

```diff
@@ -0,0 +1,115 @@
+import { FunctionComponent, useEffect, useState } from "react";
+import { Redirect } from "react-router-dom";
+import {
+  Chat,
+  Issues,
+  LobbyGameTitle,
+  ResultOnCards,
+  ScrumMaster,
+  Timer,
+} from "../../../../components";
+import {
+  AppSettings,
+  GameSettingsCurrent,
+  IssuesRedux,
+  Players,
+  ResultRedux,
+  useAppSelector,
+} from "../../../../redux/store";
+import { AnimeChatMount } from "../../../../lib";
+import { Player } from "../../../Socket/types";
+import { ProgressBar } from "../ProgressBar";
+import { MemberControll } from "../MemberControlls";
+import { StopGameBtn } from "../StopGameBtn";
+import { StartGameBtn } from "../StartGameBtn";
+import { GameControll } from "../GameControll";
+import { GameCards } from "../GameCards";
+import "./style.scss";
+
+export const Layout: FunctionComponent = (): JSX.Element => {
+  const [playerRole, setPlayerRole] = useState<string>("Observer");
+  const [issueId, setIssueId] = useState<string>("");
+  const { chatOpen, tikTak, roundControl } = useAppSelector(AppSettings);
+  const playerFromRedux = useAppSelector(Players);
+  const { appStage, lobbyId, isTimerNeed } =
+    useAppSelector(GameSettingsCurrent);
+  const issueResults = useAppSelector(ResultRedux);
+  const { issues } = useAppSelector(IssuesRedux);
+
+  useEffect(() => {
+    const player = sessionStorage.getItem("player");
+    if (player) {
+      const { role } = JSON.parse(player) as Player;
+      setPlayerRole(role);
+    }
+  }, [playerRole]);
+
+  useEffect(() => {
+    const votingIssueIdx = issues.findIndex(
+      (issue) => issue.issueStatus === "voting"
+    );
+    console.log(votingIssueIdx);
+    if (votingIssueIdx >= 0) {
+      const curIssue = issues[votingIssueIdx];
+      setIssueId(curIssue.id);
+    }
+  }, [issues, issueResults]);
+
+  if (appStage === "results") {
+    return <Redirect to={`/result/${lobbyId}`} />;
+  }
+
+  return (
+    <>
```

Comment:
again - big component which can be split at least into container and view components, for the view component - it can be split into smaller ones as well